tests/test_pytypes.py

```diff
@@ -583,6 +584,31 @@ def callback(wr):
     assert callback_called
 
 
+@pytest.mark.parametrize(
+    "create_weakref, has_callback",
+    [
+        (m.weakref_from_handle, False),
+        (m.weakref_from_object, False),
+        (m.weakref_from_handle_and_function, True),
+        (m.weakref_from_object_and_function, True),
+    ],
+)
+def test_weakref_err(create_weakref, has_callback):
+    class C:
+        __slots__ = []
+
+    def callback(_):
+        pass
+
+    ob = C()
+    # Should raise TypeError on CPython
+    with pytest.raises(TypeError) if not env.PYPY else contextlib.nullcontext():
+        if has_callback:
+            _ = create_weakref(ob, callback)
```

Comment:
Is the `_` important here? (Guessing yes, but asking JIC it's a leftover from some experiment.)

Reply:
I just wanted to annotate that there was a return value we were throwing away.

Reply:
Makes sense! Thanks for the explanation.